dev/src/index.ts

```diff
@@ -592,7 +595,8 @@ export class Firestore {
           {optional: true, minValue: 1});
     }
 
-    return this._runTransaction(updateFunction, transactionOptions);
+    return this.initializeIfNeeded().then(
```

Comment:
Yes, this is intentional. Async functions do not throw exceptions, instead they return a failed Promise. Since we want to validate our input synchronously, I did not use async/await in functions that perform validation.